README.md

```diff
@@ -41,56 +42,58 @@ OpenChatML is intended to be used in various conversational AI scenarios, includ
 4. Human-in-the-Loop Interactions: Representing and capturing interactions between human agents and AI systems for training and evaluation.
 
 ## 2. Tokens
+
 OpenChatML uses the following special tokens:
 
-- `<s>`: Beginning of Sequence (BOS) token, indicating the start of a conversation.
-- `</s>`: End of Sequence (EOS) token, indicating the end of the conversation.
-- `<|im_start|>`: Start of Turn token, indicating the beginning of a new message within the conversation. 
+- `<|BOS|>`: Beginning of Sequence (BOS) token, indicating the start of a conversation.
+- `<|EOS|>`: End of Sequence (EOS) token, indicating the end of the conversation.
+- `<|im_start|>`: Start of Turn token, indicating the beginning of a new message within the conversation.
 - `<|im_end|>`: End of Turn token, indicating the end of the current message.
 - `<|fim_prefix|>`: Before Cursor token, indicating the content before the cursor in a fill-in-the-middle task.
 - `<|fim_suffix|>`: After Cursor token, indicating the content after the cursor in a fill-in-the-middle task.
 - `<|fim_middle|>`: At Cursor token, indicating where the model should fill in content in a fill-in-the-middle task.
 - `<|file_separator|>`: File Separator token, used to separate content from different files within the same sequence.
-- `<|startofthought|>`: Start of Thought token, indicating the beginning of a thought or rationale.
-- `<|endofthought|>`: End of Thought token, indicating the end of a thought or rationale.
+- `<|start_of|>thought`: Start of Thought token, indicating the beginning of a thought or rationale.
```

Comment:
Isn't this breaking the consistency of the special tokens this way?

Reply:
> Isn't this breaking the consistency of the special tokens this way?

The PR is based on https://github.com/cognitivecomputations/OpenChatML/issues/4, specifically "Thought Structure":

> Also consider emotions and actions, those are another layer that could deserve its own tags (or a general tag with a specific qualifier), as we should support the AI outputting emotional states and real or simulated actions in an independent structure from the actual message, e. g. to have TTS use the emotion as a generation parameter of *how* to speak, without explicitly saying the emotional state out loud. Or actions, which are otherwise often asterisk-delimited, but would be more useful and less ambiguous to have in a clearly defined format (and for video generation/VR/robot control, that might later also be clearly separated from the actual written/spoken text). That's why I suggest `<|start_of|>…` / `<|end_of|>…`.

I thought having a universal tag to start and end special messages like thoughts, emotions, actions, etc. would make sense. However, after having seen the number of special tokens Meta reserved in the Llama 3 Instruct tokenizer, I now think it would be better to just have a bunch of special tokens instead of universal ones plus regular strings. Would save tokens, prevent the string influencing output in unintended ways, and be more readable.

Reply:
yep i think better to have extra tokens